core/sys/linux/sys.odin

```diff
@@ -216,7 +216,17 @@ rt_sigprocmask :: proc "contextless" (mask_kind: Sig_Mask_Kind, new_set: ^Sig_Se
 	return Errno(-ret)
 }
 
-// TODO(flysand): ioctl
+/*
+	Control devices. The ioctl syscall is a bit special because
+	its argument is usually a pointer to some driver-specific structure.
+	The request value is device-specific. Consult your LibC implementation's
+	ioctls.h file to learn more.
+	Available since Linux 1.0.
+*/
+ioctl :: proc "contextless" (fd: Fd, request: u64, arg: u64) -> (Errno) {
+	ret := syscall(SYS_ioctl, fd, request, arg)
+	return Errno(-ret)
```

Comment:
Yeah this is one of the reasons I didn't want to do the syscall. Some of ioctl commands return a value, while others don't. I don't feel like bothering making enums and proc groups for this particular syscall is going to be worth it. Maybe splitting it up into groups like [`ioctl_console`](https://man7.org/linux/man-pages/man2/ioctl_console.2.html) and [`ioctl_fat`](https://man7.org/linux/man-pages/man2/ioctl_fat.2.html) (like man pages do) is a better strategy for making the API somewhat sane. It's a ton of work though

Reply:
The function signature has been adjusted, I forgot that `man` documentation is straight up wrong sometimes.

The problem with `ioctl` is that it is designed to be extremely flexible, I'm not sure if there is a saner way besides splitting it into many groups and copying all the request codes, and it would still not be complete as some calls are vendor-specific. The main reason I introduced this binding was to do some terminal IO that couldn't be done with just POSIX bindings.